src/openms/source/FORMAT/MRMFeatureQCFile.cpp

```diff
@@ -28,265 +28,180 @@
+    it = headers.find("overall_quality_l");
+    cg.overall_quality_l = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("overall_quality_u");
+    cg.overall_quality_u = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("n_heavy_l");
+    cg.n_heavy_l = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_heavy_u");
+    cg.n_heavy_u = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_light_l");
+    cg.n_light_l = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_light_u");
+    cg.n_light_u = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_detecting_l");
+    cg.n_detecting_l = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_detecting_u");
+    cg.n_detecting_u = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_quantifying_l");
+    cg.n_quantifying_l = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_quantifying_u");
+    cg.n_quantifying_u = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_identifying_l");
+    cg.n_identifying_l = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_identifying_u");
+    cg.n_identifying_u = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_transitions_l");
+    cg.n_transitions_l = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("n_transitions_u");
+    cg.n_transitions_u = it != headers.end() ? line[it->second].toInt() : 0;
+    it = headers.find("ion_ratio_pair_name_1");
+    cg.ion_ratio_pair_name_1 = it != headers.end() ? line[it->second] : "";
+    it = headers.find("ion_ratio_pair_name_2");
+    cg.ion_ratio_pair_name_2 = it != headers.end() ? line[it->second] : "";
+    it = headers.find("ion_ratio_l");
+    cg.ion_ratio_l = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("ion_ratio_u");
+    cg.ion_ratio_u = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("ion_ratio_feature_name");
+    cg.ion_ratio_feature_name = it != headers.end() ? line[it->second] : "";
+
+    for (const std::pair<String, Size>& h : headers) // parse the parameters
     {
-      // split into meta_value_key and lub
-      // example "meta_value_value_l" -> "meta_value_value" and "l"
-      meta_value_key = kv.first.substr(0, kv.first.length()-2);
-      lub = kv.first.substr(kv.first.length()-1, kv.first.length());
-      if (cqcs.meta_value_qc.count(meta_value_key) == 0)
-      {     
-        cqcs.meta_value_qc[meta_value_key] = lbub;
-      }
-        
-      // cast doubles
-      if (lub == "l")
+      const String& header = h.first;
+      const Size& i = h.second;
+      boost::smatch m;
+      if (boost::regex_search(header, m, boost::regex("metaValue_(.+)_(l|u)")))
```

Comment:
Perhaps a few comments on what how this implementation is capturing the lower and upper bound columns might be helpful.